packages/gamut/src/GridForm/index.tsx

```diff
@@ -76,7 +88,10 @@ export function GridForm<
             />
           );
         })}
-        <GridFormSubmit {...submit} />
+        <GridFormSubmit
+          disabled={validation === 'onChange' ? !formState.isValid : false}
```

Comment:
Does this work? Boolean logic always takes a bit to kick in before lunch for me...

```suggestion
          disabled={validation === 'onChange' && !formState.isValid}
```